Update Karpenter usage documentation to reflect latest supported version(s) (#8250)
eksctl-io/eksctl: userdocs/src/usage/eksctl-karpenter.md

`eksctl` provides adding [Karpenter](https://karpenter.sh/) to a newly created cluster. It will create all the necessary
prerequisites outlined in Karpenter's [Getting Started](https://karpenter.sh/docs/getting-started/) section including installing
Karpenter itself using Helm. We currently support installing versions starting `0.28.0` and above. See in The [Compatibility](https://karpenter.sh/docs/upgrading/compatibility/) section.

> ???+ info

Reviewer: lets just remove this whole block now - lines 7-10

    With [v0.17.0](https://karpenter.sh/docs/upgrading/upgrade-guide/) Karpenter’s Helm chart package is now stored in Karpenter’s OCI (Open Container Initiative) registry.
    Clusters created on previous versions shouldn't be affected by this change. If you wish to upgrade your current installation of Karpenter please refer to the [upgrade guide](https://karpenter.sh/docs/upgrading/upgrade-guide/)
    You have to be logged out of ECR repositories to be able to pull the OCI artifact by running `helm registry logout public.ecr.aws` or `docker logout public.ecr.aws`, failure to do so will result in a 403 error when trying to pull the chart.

> 

Reviewer: Lets change the next line to simply:
```
The following yaml outlines a typical installation configuration:
```

```yaml
karpenter:
  version: '1.2.1'
  createServiceAccount: true # default is false
  defaultInstanceProfile: 'KarpenterNodeInstanceProfile' # default is to use the IAM instance profile created by eksctl
  withSpotInterruptionQueue: true # adds all required policies and rules for supporting Spot Interruption Queue, default is false
```

OIDC must be defined in order to install Karpenter.

> Once Karpenter is successfully installed, add a [NodePools](https://karpenter.sh/docs/concepts/nodepools/) and [NodeClasses](https://karpenter.sh/docs/concepts/nodeclasses/) so Karpenter

Reviewer: ```suggestion
Once Karpenter is successfully installed, add [NodePool(s)](https://karpenter.sh/docs/concepts/nodepools/) and [NodeClass(es)](https://karpenter.sh/docs/concepts/nodeclasses/) so Karpenter
```

can start adding the right nodes to the cluster.

The NodePool's `nodeClassRef` section must match the created `EC2NodeClass` metadata name. For example:

```yaml
apiVersion: karpenter.sh/v1
kind: NodePool
metadata:

>   name: general-purpose

Reviewer: lets not call it this due to Auto Mode
```suggestion
  name: default
```

  annotations:

>     kubernetes.io/description: "General purpose NodePool for generic workloads"

Reviewer: ```suggestion
    kubernetes.io/description: "General purpose NodePool"
```

spec:
  template:
    spec:
      requirements:
        - key: kubernetes.io/arch
          operator: In
          values: ["amd64"]
        - key: kubernetes.io/os
          operator: In
          values: ["linux"]
        - key: karpenter.sh/capacity-type
          operator: In
          values: ["on-demand"]
        - key: karpenter.k8s.aws/instance-category
          operator: In
          values: ["c", "m", "r"]
        - key: karpenter.k8s.aws/instance-generation
          operator: Gt
          values: ["2"]
      nodeClassRef:
        group: karpenter.k8s.aws
        kind: EC2NodeClass

>         name: default

Reviewer: ```suggestion
        name: example
```

> 

Reviewer: remove blank line

```

```yaml
apiVersion: karpenter.k8s.aws/v1
kind: EC2NodeClass
metadata:
  name: default
  annotations:

>     kubernetes.io/description: "General purpose EC2NodeClass for running Amazon Linux 2 nodes"

Reviewer: ```suggestion
    kubernetes.io/description: "Default EC2NodeClass"
```
